Beam Receipt NPE

With all the changes, the beam support is now broken on receipt. Trying to launch the application on receipt of the beam causes an NPE.

```
FATAL EXCEPTION: main
Process: net.somethingdreadful.MAL, PID: 21481
java.lang.RuntimeException: Unable to start activity ComponentInfo{net.somethingdreadful.MAL/net.somethingdreadful.MAL.DetailView}: java.lang.NullPointerException
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2184)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2233)
    at android.app.ActivityThread.access$800(ActivityThread.java:135)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:136)
    at android.app.ActivityThread.main(ActivityThread.java:5001)
    at java.lang.reflect.Method.invokeNative(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:515)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
    at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
    at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
    at net.somethingdreadful.MAL.DetailView.setCard(DetailView.java:170)
    at net.somethingdreadful.MAL.DetailView.onCreate(DetailView.java:87)
    at android.app.Activity.performCreate(Activity.java:5231)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2148)
    ... 12 more
```

--- a/Atarashii/src/net/somethingdreadful/MAL/DetailView.java
+++ b/Atarashii/src/net/somethingdreadful/MAL/DetailView.java
@@ -167,7 +167,7 @@ public class DetailView extends Activity implements Serializable, OnRatingBarCha
      * Manage the progress card
      */
     public void setCard() {
-        if (type.equals(ListType.ANIME)) {
+        if (type != null && type.equals(ListType.ANIME)) {
             TextView progresslabel1 = (TextView) findViewById(R.id.progresslabel1);
             progresslabel1.setText(getString(R.string.card_content_episodes));
             TextView progresslabel2 = (TextView) findViewById(R.id.progresslabel2);
@@ -266,9 +266,11 @@ public class DetailView extends Activity implements Serializable, OnRatingBarCha
      * Get the records (Anime/Manga)
      */
     public void getRecord() {
-        Bundle data = new Bundle();
-        data.putInt("recordID", recordID);
-        new NetworkTask(TaskJob.GET, type, context, data, this, this).execute();
+        if (type != null) {
+            Bundle data = new Bundle();
+            data.putInt("recordID", recordID);
+            new NetworkTask(TaskJob.GET, type, context, data, this, this).execute();
+        }
     }
 
     /*
@@ -460,6 +462,8 @@ public class DetailView extends Activity implements Serializable, OnRatingBarCha
      * Place all the text in the right textview
      */
     public void setText() {
+        if (type == null || (animeRecord == null && mangaRecord == null)) // not enough data to do anything
+            return;
         GenericRecord record;
         setMenu();
         TextView status = (TextView) findViewById(R.id.cardStatusLabel);

